pandas/core/arrays/period.py

```diff
@@ -465,7 +472,13 @@ def _time_shift(self, n, freq=None):
     @property
     def _box_func(self):
         # Used in DatelikeArray.__iter__
-        return lambda x: Period._from_ordinal(ordinal=x, freq=self.freq)
+        # TODO: implement this in cython?
+        def func(x):
+            if isinstance(x, Period) or x is NaT:
```

Comment:
Where does it occur that `x` is already a Period? 
That seems that `_box_func` is then used in two very different circumstances?

Reply:
Ah, I see this was coming from the Index implementation

Reply:
I'm looking into this now. I feel like we shouldn't have to change it.

Reply:
Ahh I see.

So if I do something basic like `period_index[0]` `DatetimelikeIndex.__getitem__` does

```
    169         getitem = self._data.__getitem__
    170         if is_int:
    171             val = getitem(key)
--> 172             return self._box_func(val)
 ```

`getitem` is `PeriodArray.__getitem__`, which has to already box ordinals as Periods (to fit the EA API). So we try to rebox it.

I was hoping to simplify by moving these to a single `_box_func`, but I think this is more complicated.